### PR TITLE
scheduler: add leader verify in balance-region scheduler (#2966)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #2966 to release-4.0

---

Signed-off-by: Song Gao <disxiaofei@163.com>

### What problem does this PR solve?
If the region have no leader during balance-region, it will pause PD panic and restart over and over.

### What is changed and how it works?
Check region leader before balance region. If it has no leader, the balance for this region would be skipped.

### Check List

Tests

- Unit test

Related changes
- Need to cherry-pick to the release branch

### Release note

- Fix the bug that pd might panic If some regions have no leader when `balance-region` enabled.

### CODE REVIEW

**Review by lhy1024 — APPROVED** [review]

**Review by Yisaer — APPROVED** [review]
LGTM

File: server/schedulers/balance_test.go [hunk 1]
```diff
@@ -1125,6 +1125,23 @@ func (s *testReplicaCheckerSuite) TestOpts(c *C) {
 	c.Assert(rc.Check(region), IsNil)
 }
 
+func (s *testBalanceRegionSchedulerSuite) TestShouldNotBalance(c *C) {
+	opt := config.NewTestOptions()
+	tc := mockcluster.NewCluster(opt)
+	tc.DisableFeature(versioninfo.JointConsensus)
```
Comment (HunDunDM) [inline server/schedulers/balance_test.go:1131]: not supported in `release-4.0`

Comment (Yisaer) [reply server/schedulers/balance_test.go]: good catch

**Pull request merged**